File: app/src/main/java/com/ritesh/imagetopdf/db/PdfDao.java
package com.ritesh.imagetopdf.db;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface PdfDao {

    @Insert
    void insert(PdfDetail pdfDetail);

    @Update
    void update(PdfDetail pdfDetail);

    @Delete
    void delete(PdfDetail pdfDetail);

    @Query("SELECT * FROM PdfDetail ORDER BY id DESC")
    LiveData<List<PdfDetail>> getAllPdfs();

}

File: app/src/main/java/com/ritesh/imagetopdf/data/PdfDao.kt
package com.ritesh.imagetopdf.data

import androidx.room.*
import kotlinx.coroutines.flow.Flow

@Dao
interface PdfDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(pdfEntity: PdfEntity)

    @Delete
    suspend fun delete(pdfEntity: PdfEntity)

    @Delete
    suspend fun delete(pdfEntityList: List<PdfEntity>)

    @Query("SELECT * FROM pdf WHERE id = :id")
    suspend fun getPdfById(id: Long): PdfEntity?

    @Query("SELECT * FROM pdf ORDER BY id DESC")
    fun getAllPdfs(): Flow<List<PdfEntity>>

    @Query("UPDATE pdf SET fileName = :name WHERE id = :id")
    suspend fun renameFile(name: String, id: Long)
}